### ISSUE TITLE
Host blows up during assembly scanning

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus.Host#115]
Since NServiceBus core Version 6 now can return partially scannable assemblies the host will blow up since it does its own `.GetTypes()` call on each assembly in order to find the class implementing `IConfigureThisEndpoint`.

### Symptoms

The Version 7 host blows up during startup if there is a assembly that is partially scannable. This forces the users to add assembly redirects to both `NServiceBus.Host.exe.config` and `app.config`

### Who's affected

All version 7 users.

### Workaround

Add a `NServiceBus.Host.exe.config` file with the same redirects as the app.config

Replaces #114

**Comment by SimonCropp** [comment]
should this have a milestone?

**Comment by SeanFeldman** [comment]
In addition to @SimonCropp `s question, should a bug issue have [proper sections](https://github.com/Particular/PlatformDevelopment/blob/master/guidelines/release-notes.md#bugs) to appear in the release notes>

**Comment by adamralph** [comment]
ping @Particular/host-maintainers

**Comment by sfarmar** [comment]
@ramonsmits Let's have a quick sync next week?

**Comment by ramonsmits** [comment]
@andreasohlund So why was this an issue? Why do we do not want the previous behavior? If some kind of binding redirect is incorrect or an assembly is missing wouldn't we *want* to blow up? The google group post also indicates that the issue was a missing assembly that made the project work after adding it.

**Comment by andreasohlund** [comment]
>So why was this an issue? Why do we do not want the previous behavior? If some kind of binding redirect is incorrect or an assembly is missing wouldn't we want to blow up? 

The host has always done a best effort scanning to find the endpoint config (and throw if not found). If the scanner issues isn't not resolved by redirects in app.config the endpoint would throw when it starts instead.

This saves the user from having to put redirect in both `NServiceBus.Host.exe.config` and their own app.config (not that both we and nuget actually auto adds the correct redirects to app.config)

In short: To not annoy users we should not throw in the "outer scan"

Does this make sense?

**Comment by ramonsmits** [comment]
@andreasohlund Ok, got it, it is specific in resolving the `IConfigureThisEndpoint` to check if it can be located although other scanned types could not be resolved.

That makes sense, especially if there are types that would never be used at runtime at all.

**Comment by adamralph** [comment]
So should this be added to a 7.1.0 milestone?
